src/mame/drivers/hh_sm510.cpp

```diff
@@ -1137,6 +1137,90 @@ MACHINE_CONFIG_END
+  * vertical dual lcd screens with custom segments, 1-bit sound
+
+***************************************************************************/
+
+class gnw_dkong_state : public hh_sm510_state
+{
+public:
+        gnw_dkong_state(const machine_config &mconfig, device_type type, const char *tag)
+                : hh_sm510_state(mconfig, type, tag)
+        {
+                m_inp_lines = 3;
+        }
+
+        void gnw_dkong(machine_config &config);
+};
+
+// config
+
+static INPUT_PORTS_START( gnw_dkong )
+        PORT_START("IN.0") // S1
+        PORT_BIT( 0x07, IP_ACTIVE_HIGH, IPT_UNUSED )
+        PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, input_changed, nullptr) // Jump
+
+        PORT_START("IN.1") // S2
+        PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_JOYSTICK_RIGHT ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, input_changed, nullptr)
+        PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_JOYSTICK_UP ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, input_changed, nullptr)
+        PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_JOYSTICK_LEFT ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, input_changed, nullptr)
+        PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_JOYSTICK_DOWN ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, input_changed, nullptr)
+
+        PORT_START("IN.2") // S3
+        PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_SELECT ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, input_changed, nullptr) PORT_NAME("Time")
+        PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_START2 ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, input_changed, nullptr) PORT_NAME("Game B")
+        PORT_BIT( 0x04, IP_ACTIVE_HIGH, IPT_START1 ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, input_changed, nullptr) PORT_NAME("Game A")
+        PORT_BIT( 0x08, IP_ACTIVE_HIGH, IPT_SERVICE2 ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, input_changed, nullptr) PORT_NAME("Alarm")
+
+        PORT_START("ACL")
+        PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_SERVICE1 ) PORT_CHANGED_MEMBER(DEVICE_SELF, hh_sm510_state, acl_button, nullptr) PORT_NAME("ACL")
+
+        PORT_START("B") // MCU B(beta) pin pulled to GND
+        PORT_CONFNAME( 0x01, 0x01, "Infinite Lives (Cheat)")
+        PORT_CONFSETTING(    0x01, DEF_STR( Off ) )
+        PORT_CONFSETTING(    0x00, DEF_STR( On ) )
+INPUT_PORTS_END
+
+MACHINE_CONFIG_START(gnw_dkong_state::gnw_dkong)
+
+        /* basic machine hardware */
+        MCFG_DEVICE_ADD("maincpu", SM510)
+        MCFG_SM510_R_MASK_OPTION(2) // confirmed
+        MCFG_SM510_WRITE_SEGS_CB(WRITE16(*this, hh_sm510_state, sm510_lcd_segment_w))
+        MCFG_SM510_READ_K_CB(READ8(*this, hh_sm510_state, input_r))
+        MCFG_SM510_WRITE_S_CB(WRITE8(*this, hh_sm510_state, input_w))
+        MCFG_SM510_WRITE_R_CB(WRITE8(*this, hh_sm510_state, piezo_r1_w))
+        MCFG_SM510_READ_B_CB(IOPORT("B"))
+
+        /* video hardware */
+        MCFG_SCREEN_SVG_ADD("screen_top", "svg_top")
```

Comment:
Aspect ratio here is copy-pasted from the dkong2 driver.
If the SVGs are the same size as the ones you sent me earlier today, height for both screens is 1266/2.

Reply:
Thanks for finding this! I've included this fix in a new commit :)